string targetAddress
uint16 targetPort
string sendMessageTopicName
bool shouldListen
string receiveAddress
uint16 receivePort
string receiveMessageTopicName
uint8 receiveThreadCount
---
bool result